pkg/join/join.go

```diff
@@ -68,17 +68,15 @@ func ClusterToBroker(brokerInfo *broker.Info, options *Options, clientProducer c
 		ClusterSize: options.GlobalnetClusterSize,
 	}
 
-	if options.GlobalnetEnabled {
-		status.Start("Discovering multi cluster details")
+	status.End()
 
-		err = globalnet.AllocateAndUpdateGlobalCIDRConfigMap(options.ClusterID, brokerAdminClientset, brokerNamespace, &netconfig)
+	if options.GlobalnetEnabled {
+		err = globalnet.AllocateAndUpdateGlobalCIDRConfigMap(brokerAdminClientset, brokerNamespace, &netconfig, status)
 		if err != nil {
-			return status.Error(err, "Error Discovering multi cluster details")
+			return err // nolint:wrapcheck // No need to wrap here
```

Comment:
I think a high-level error message is good to have here. and then `globalnet.go` has detailed error specified.

Reply:
I'm not sure I follow. Do you mean wrap the return error or add report it to the status? For the latter, `globalnet.AllocateAndUpdateGlobalCIDRConfigMap` already does that.

Reply:
I assume you meant the former - it's not wrapped.

Reply:
I meant to keep the writeup `"Error Discovering multi cluster details"` and not delete it.

Reply:
Frankly, I don't think that message really makes sense in this context. I changed it to "_unable to determine the global CIDR_" which is what this code actually does.